src/api/app/controllers/webui/users_controller.rb

```diff
@@ -11,13 +11,37 @@ def index
   end
 
   def show
-    @iprojects = @displayed_user.involved_projects.pluck(:name, :title)
-    @ipackages = @displayed_user.involved_packages.joins(:project).pluck(:name, 'projects.name as pname')
-    @owned = @displayed_user.owned_packages
     @groups = @displayed_user.groups
     @role_titles = @displayed_user.roles.global.pluck(:title)
     @account_edit_link = CONFIG['proxy_auth_account_page']
 
+    if Flipper.enabled?(:user_profile_redesign, User.possibly_nobody)
+      filters = params.slice(:search_text, :involved_projects, :involved_packages,
+                             :role_owner, :role_maintainer, :role_bugowner, :role_reviewer, :role_downloader, :role_reader)
+
+      filters[:search_text] = filters[:search_text]&.strip
+      @filters = filters.dup
+
+      if !filters.key?(:involved_projects) && !filters.key?(:involved_packages)
+        [:involved_projects, :involved_packages].each do |filter|
+          filters[filter] = 1
+        end
+      end
+      if !filters.key?(:role_owner) && !filters.key?(:role_maintainer) && !filters.key?(:role_bugowner) &&
+         !filters.key?(:role_reviewer) && !filters.key?(:role_downloader) && !filters.key?(:role_reader)
+        [:role_owner, :role_maintainer, :role_bugowner, :role_reviewer, :role_downloader, :role_reader].each do |filter|
+          filters[filter] = 1
+        end
+      end
```

Comment:
How about moving this code into a method? Something like `filters_for_involved_items` or else.

Reply:
Codeclimate was complaining about it. I was moving the code already, to `adjust_filters`. It should be solved. Please have a look again.